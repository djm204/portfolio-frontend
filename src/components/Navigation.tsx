import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import cnames from 'classnames';
import '../stylesheets/Egg.css';


const Navigation = () => {
    const [fixedNav, setFixedNav] = useState();

    window.addEventListener('scroll', () => {
        if(window.scrollY > window.innerHeight-20 && !fixedNav) {
            setFixedNav(true);
        }
        
        if(window.scrollY < window.innerHeight-20 && fixedNav){
            setFixedNav(false);
        }
    });

    const goToSection = (event:  React.MouseEvent<HTMLAnchorElement | MouseEvent>) => {
        event.preventDefault();
        const et = event.currentTarget as HTMLAnchorElement;
        const target = document.querySelector(`#${et.dataset.goto}`);
        if (target) {
            target.scrollIntoView({
                behavior: 'smooth'
            });
        }
    }

    return (
        <React.Fragment>
            <nav className={cnames('Nav', { 'Nav__Fixed': fixedNav })}>
                <a onClick={goToSection} href="/about" data-goto="About">About</a>
                <a onClick={goToSection} href="/tech-summary" data-goto="TechSummary">Tech Summary</a>
                <a onClick={goToSection} href="/projects" data-goto="Projects">Projects</a>
                <a onClick={goToSection} href="/contact" data-goto="Contact">Contact</a>
                <div className="EggBert">
                    <FontAwesomeIcon icon={'egg'} />
                </div>
            </nav>
        </React.Fragment>
    );
}

export default Navigation;
