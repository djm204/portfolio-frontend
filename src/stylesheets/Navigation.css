.Nav {
  background: rgba(0, 34, 111, 1);
  display: flex;
  z-index: 99999999;
  letter-spacing: 0.09em;
  min-height: 48.25px;
}

.Nav__Links {
  display: flex;
}

.Nav a {
  color: rgba(172, 197, 255, 1);
  padding: 10px 15px;
}

.Nav a:hover {
  color: rgba(216, 144, 0, 1);
  text-decoration: none;
  background: rgba(26, 95, 255, 0.6);
}

.Nav__Fixed {
  position: fixed;
  width: 100%;
  animation: popDown ease-out 0.5s;
}

.Hamburger__MobileNav {
  color: white;
  font-size: 1.7em;
  padding: 5px;
}

.Nav__Mobile .Nav__Links {
  flex-direction: column;
  width: 100%;
}

.Nav.Nav__Mobile .EggBert {
  width: 100%;
}

.Nav__Hidden {
  display: none;
}
