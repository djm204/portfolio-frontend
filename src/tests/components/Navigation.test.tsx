import React from 'react'
import Navigation from '../../components/Navigation'
import { HashRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'

it('Navigation renders without crashing', () => {
  const result = render(
    <Router>
      <Navigation />
    </Router>
  )

  expect(result).toMatchSnapshot()
})
