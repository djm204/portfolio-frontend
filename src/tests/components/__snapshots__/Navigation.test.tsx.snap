// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="Nav"
      >
        <div
          class="Hamburger__MobileNav Nav__Hidden"
        >
          <i
            class="fa"
          />
        </div>
        <div
          class="Nav__Links"
        >
          <a
            data-goto="About"
            href="/about"
          >
            About
          </a>
          <a
            data-goto="TechSummary"
            href="/tech-summary"
          >
            Tech Summary
          </a>
          <a
            data-goto="Projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            data-goto="Contact"
            href="/contact"
          >
            Contact
          </a>
          <button
            class="EggBert"
            role="show-eggs"
          >
            <i
              class="fa"
            />
          </button>
          <div
            class="Egg Egg--1 Egg__Hide"
            data-testid="egg"
          >
            <div
              class="image-wrapper"
            >
              <img
                alt="I love to festival"
                src="https://davidmendez.dev/i_love_to_festival.jpg"
              />
            </div>
          </div>
          <div
            class="Egg Egg--2 Egg__Hide"
            data-testid="egg"
          >
            <div
              class="image-wrapper"
            >
              <img
                alt="I love to garden"
                src="https://davidmendez.dev/i_love_to_garden.jpg"
              />
            </div>
          </div>
          <div
            class="Egg Egg--3 Egg__Hide"
            data-testid="egg"
          >
            <div
              class="image-wrapper"
            >
              <img
                alt="I love my cats"
                src="https://davidmendez.dev/i_love_my_cats.jpg"
              />
            </div>
          </div>
          <div
            class="Egg Egg--4 Egg__Hide"
            data-testid="egg"
          >
            <div
              class="image-wrapper"
            >
              <img
                alt="My love and I"
                src="https://davidmendez.dev/me_and_my_love.jpg"
              />
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="Nav"
    >
      <div
        class="Hamburger__MobileNav Nav__Hidden"
      >
        <i
          class="fa"
        />
      </div>
      <div
        class="Nav__Links"
      >
        <a
          data-goto="About"
          href="/about"
        >
          About
        </a>
        <a
          data-goto="TechSummary"
          href="/tech-summary"
        >
          Tech Summary
        </a>
        <a
          data-goto="Projects"
          href="/projects"
        >
          Projects
        </a>
        <a
          data-goto="Contact"
          href="/contact"
        >
          Contact
        </a>
        <button
          class="EggBert"
          role="show-eggs"
        >
          <i
            class="fa"
          />
        </button>
        <div
          class="Egg Egg--1 Egg__Hide"
          data-testid="egg"
        >
          <div
            class="image-wrapper"
          >
            <img
              alt="I love to festival"
              src="https://davidmendez.dev/i_love_to_festival.jpg"
            />
          </div>
        </div>
        <div
          class="Egg Egg--2 Egg__Hide"
          data-testid="egg"
        >
          <div
            class="image-wrapper"
          >
            <img
              alt="I love to garden"
              src="https://davidmendez.dev/i_love_to_garden.jpg"
            />
          </div>
        </div>
        <div
          class="Egg Egg--3 Egg__Hide"
          data-testid="egg"
        >
          <div
            class="image-wrapper"
          >
            <img
              alt="I love my cats"
              src="https://davidmendez.dev/i_love_my_cats.jpg"
            />
          </div>
        </div>
        <div
          class="Egg Egg--4 Egg__Hide"
          data-testid="egg"
        >
          <div
            class="image-wrapper"
          >
            <img
              alt="My love and I"
              src="https://davidmendez.dev/me_and_my_love.jpg"
            />
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
